mutation/partition_version.cc

```diff
@@ -450,12 +450,12 @@ utils::coroutine partition_entry::apply_to_incomplete(const schema& s,
                 bool static_row_continuous = dst_snp->static_row_continuous();
                 auto current = &*src_snp->version();
                 while (current) {
-                    dirty_size += allocator.object_memory_size_in_allocator(current)
-                        + current->partition().static_row().external_memory_usage(s, column_kind::static_column);
                     dst.partition().apply(current->partition().partition_tombstone());
                     if (static_row_continuous) {
                         lazy_row& static_row = dst.partition().static_row();
                         if (can_move) {
+                            dirty_size += allocator.object_memory_size_in_allocator(current)
+                                + current->partition().static_row().external_memory_usage(s, column_kind::static_column);
```

Comment:
I'm ambivalent about this change. On one hand it is correct in reflecting the reality, if !can_move, we didn't release any memory. On the other hand, if there is a large partition snapshot in the memtable, real dirty will not be decreased until the snapshot dies, and can cause write latency spikes. Current code trades lower latency at the risk of OOMing. Refs  https://github.com/scylladb/scylladb/issues/1938.